test/replication-luatest/gh_9918_synchro_queue_additional_info_test.lua

```diff
@@ -100,3 +100,72 @@ g.test_age_field = function(cg)
         t.assert_equals(box.info.synchro.queue.age, 0)
     end)
 end
+
+-- Test that new `confirm_lag` field of `box.info.synchro.queue` works
+-- correctly.
+g.test_confirm_lag_field = function(cg)
+    -- The were no synchronous transactions since the start.
+    cg.leader:exec(function()
+        t.assert_equals(box.info.synchro.queue.confirm_lag, 0)
+    end)
+    cg.replica:exec(function()
+        t.assert_equals(box.info.synchro.queue.confirm_lag, 0)
+    end)
+
+    -- Ensure that the synchronous transaction stays in the queue for a while.
+    cg.leader:update_box_cfg{replication_synchro_quorum = 3,
+                             replication_synchro_timeout = 60}
+    -- Add an entry to the synchronous queue.
+    local fid, oldest_entry_ts = cg.leader:exec(function()
+        local fiber = require('fiber')
+
+        local f = fiber.new(function() box.space.s:replace{0} end)
+        f:set_joinable(true)
+        return f:id(), fiber.clock()
+    end)
+    fiber.sleep(1)
+
+    -- Nothing has been confirmed yet.
+    cg.leader:exec(function()
+        t.assert_equals(box.info.synchro.queue.confirm_lag, 0)
+    end)
+    cg.replica:exec(function()
+        t.assert_equals(box.info.synchro.queue.confirm_lag, 0)
+    end)
+
+    -- Allow the synchronous queue to advance.
+    cg.leader:update_box_cfg{replication_synchro_quorum = ''}
+    cg.leader:exec(function(fid)
+       t.assert(require('fiber').find(fid):join())
+    end, {fid})
+
+    -- The confirm lag must now be shown.
+    local wait_time = fiber.clock() - oldest_entry_ts
+    cg.leader:exec(function(wait_time)
+        t.assert_almost_equals(box.info.synchro.queue.confirm_lag, wait_time,
+                               0.15)
+    end, {wait_time})
+    cg.replica:exec(function(wait_time)
+        t.assert_almost_equals(box.info.synchro.queue.confirm_lag, wait_time,
+                               0.15)
+    end, {wait_time})
+
+    local expected_lag = cg.leader:exec(function()
+        local fiber = require('fiber')
+
+        local start = fiber.clock()
+        box.space.s:replace{0}
+        return fiber.clock() - start
+    end)
+    t.assert_lt(expected_lag * 10, wait_time)
+
+    -- The confirm lag must be updated by the smaller one.
+    cg.leader:exec(function(expected_lag)
+        t.assert_almost_equals(box.info.synchro.queue.confirm_lag, expected_lag,
```

Comment:
Maybe just test, that it's smaller, then it was, when we had `quourum` 3?

Reply:
This doesn't seem to be a reliable test, so I just dropped this hunk.